torch/_dynamo/variables/builder.py

```diff
@@ -161,12 +161,14 @@ def __init__(
         self,
         tx,
         source: Source,
+        dynamic_spec=None,
```

Comment:
It feels weird that I am adding this field here, for all python Variables, where this only makes sense for TensorVariable. Something's wrong with this abstraction... 

Maybe I should minor the class structure for Source. 

```
@dataclass
class TensorSpec(VariableSpec)
     dynamic_spec: ...
```

Reply:
This is the wrong place for this - you want this alongside (or just directly on) the `value` on `__call___` below